File: 492_Construct_the_Rectangle/Construct_the_Rectangle.py
class Solution(object):
    def constructRectangle(self, area):
        """
        :type area: int
        :rtype: List[int]
        """
        l_dim = [area, 1]
        tmp_l = int(math.ceil(math.sqrt(area)))
        while tmp_l > 1:
            if area % tmp_l == 0:
                tmp_w = area//tmp_l
                l_dim = [max(tmp_l, tmp_w), min(tmp_l, tmp_w)]
                break
            tmp_l -= 1
        return l_dim

File: 492_Construct_the_Rectangle/Construct_the_Rectangle.go
func constructRectangle(area int) []int {
    ans := []int{area, 1}
    sqrt_num := math.Sqrt((float64(area)))
    sqrt_ceil := int(math.Ceil(sqrt_num))
    for i:= sqrt_ceil; i > 1; i-- {
        if area % i == 0 {
            w := area/i
            if i >= w {
                ans[0] = i
                ans[1] = w
            }else {
                ans[0] = w
                ans[1] = i
            }
            return ans
        }
    }
    return ans // worst case
}
